marc.py

```diff
@@ -521,15 +521,7 @@ class MARCExporter(object):
         },
     ]
 
-    SETTINGS = [
-        { "key": STORAGE_PROTOCOL,
-          "label": _("Storage service protocol"),
-          "description": _("Storage protocol to use for uploading generated MARC files. The service must already be configured under 'Storage Services'."),
-          "type": "select",
-          "options": [{ "key": ExternalIntegration.S3, "label": ExternalIntegration.S3 }],
-          "default": ExternalIntegration.S3,
-        },
-    ]
```

Comment:
This used to be a drop-down list where you chose which protocol to use. It needs to become a drop-down list where you choose which _integration_ to use. If you have seven different storage integrations, there need to be 7 options in this list.

This is the big challenge and I think this is why we didn't make this change earlier -- we don't have any other situation where you have to connect an ExternalIntegration to _another ExternalIntegration_. The options for this configuration setting need to be gathered at runtime, by searching for all ExternalIntegrations of the STORAGE type.

The value of this configuration setting would presumably be the database ID of the storage integration you chose. If you set this value and then delete the storage integration, I'd act as if the value was not set -- this would probably mean you can't export any MARC files until you go into the admin interface and fix it.

Reply:
I can't see very far into the future, but I can foresee a scenario where a Collection needs to designate one storage integration (an S3 integration) as the place to store its cover images, and a second storage integration (an LCP integration) as the place to store its books. So at some point we may need to have a separate database table that explains which storage integration is used as the mirror for which Collection/ExternalIntegration for which purpose.

Reply:
Does the setting creation need to be done here? I did the integration dropdown in `catalog_services` https://github.com/NYPL-Simplified/circulation/pull/1327/files#diff-cafab51e9ce4ad6b89864e1f061e3beeR23 but I did encounter of the issue of not being able to get the external integration from within this MARCExporter class.